variables/collection.go

```diff
@@ -166,6 +174,11 @@ func (c *Collection) ExitFrame() {
 	c.effectiveCache = nil
 }
 
+// IsStackAtBase returns whether the stack has size 1.
+func (c *Collection) IsStackAtBase() bool {
+	return len(c.stack) == 1
```

Comment:
nit: should we check for a condition like

    if len(c.stack) == 0 { panic("shouldnt happen") }

then again if it's empty that might cause some nil exceptions elsewhere. :shrug:

Reply:
Yes - we actually have such a check in `popStackFrame`